packages/react/src/components/List/HierarchyList/HierarchyList.jsx

```diff
@@ -496,6 +503,25 @@ const HierarchyList = ({
     }
   };
 
+  /**
+   * Used to trigger the onExpandedChange callback after the _last_ item in the expandedIds
+   * array has official expanded. Only used on regular lists, not virtual, because the virtual
+   * list only renders the rows shown, so we can't know what the _last_ item actually is.
+   *
+   * @param {string} id The ID of the item that was just expanded
+   */
+  const handleAfterExpand = (id) => {
```

Comment:
🤔 Hmm, yeah, you're right; however, even with a useCallback, it's still going to change quite frequently because of the dependencies on `expandedIds` and `previousExpandedIds`. Does that cancel out the benefits of useCallback?